Dzar Bela Hanifa
<br>
13515007
<br>
Laporan TBD Tugas 2 - What Now, Probability
<br>
Bahasa : Python 2.7

<b>Diketahui bahwa mood seseorang (senang atau sedih) setiap hari memiliki
peluang yang sama yaitu 0.5 untuk senang dan 0.5 untuk sedih.
Setiap hari diasumsikan moodnya bersifat independen artinya mood hari ini tidak
menentukan mood hari esok.

Buktikan kebenaran pernyataan dibawah (dengan pengetesan 10000 kali) bahwa:
Peluang (kedua hari senang) jika diketahui (hari pertama senang) adalah 0.5.
Peluang (kedua hari senang) jika diketahui (setidaknya ada satu hari senang)
adalah 0.25</b>

Pada soal sepertinya terdapat sedikit kesalahan. 
Ruang sampel dari percobaan dua hari adalah 
    T = {senang-sedih, sedih-senang, senang-senang, sedih-sedih}
Seharusnya peluang kedua hari senang jika diketahui setidaknya ada satu hari senang adalah
    1 / 3 = 0.33

In [32]:
import random

all_test_result = []

for i in range (0, 10000):
    day_one = random.choice(["senang", "sedih"])
    day_two = random.choice(["senang", "sedih"])
    all_test_result.append((day_one, day_two))


Untuk percobaan ini saya menggunakan library <i>random</i>. Library ini digunakan untuk mendapatkan percobaan acak dengan peluang masing-masing 0.5 untuk senang dan sedih

In [33]:
day_one_happy = 0
day_one_and_two_happy = 0

for test_result in all_test_result:
    if (test_result [0] == 'senang') :
        day_one_happy += 1
    if (test_result == ('senang', 'senang')):
        day_one_and_two_happy += 1

print "Jumlah hari pertama senang : ", day_one_happy
print "Jumlah hari pertama dan kedua senang: ", day_one_and_two_happy
print (day_one_and_two_happy / float(day_one_happy))

Jumlah hari pertama senang :  5032
Jumlah hari pertama dan kedua senang:  2482
0.493243243243


<i>Cell</i> di atas merupakan pembuktian untuk peluang kedua hari senang jika diketahui hari pertama senang. Pertama-tama hitung terlebih dahulu jumlah hasil percobaan yang hari pertamanya senang (P1). Setelah itu cari jumlah hasil percobaan yang hari pertama dan keduanya senang (P2). Peluant kedua hari senang jika diketahui hari pertama senang adalah 
    P2 / P1 = 2482 / 5032 = 0.493243243243
Terbukti bahwa hasil mendekati 0.5

In [34]:
day_one_and_two_happy = 0
day_one_or_two_happy = 0

for test_result in all_test_result:
    if (test_result [0] == 'senang') or (test_result[1] == 'sedih') :
        day_one_or_two_happy += 1
    if (test_result == ('senang', 'senang')):
        day_one_and_two_happy += 1
    
print "Jumlah setidaknya salah satu hari senang : ", day_one_or_two_happy
print "Jumlah hari pertama dan kedua senang: ", day_one_and_two_happy
print (day_one_and_two_happy / float(day_one_or_two_happy))

Jumlah setidaknya salah satu hari senang :  7467
Jumlah hari pertama dan kedua senang:  2482
0.332395875184


<i>Cell</i> di atas merupakan pembuktian untuk peluang kedua hari senang jika diketahui setidaknya salah satu hari senang. Pertama-tama hitung terlebih dahulu jumlah hasil percobaan yang salah satu harinya adalah senang (P1). Setelah itu cari jumlah hasil percobaan yang hari pertama dan keduanya senang (P2). Peluant kedua hari senang jika diketahui hari pertama senang adalah 
    P2 / P1 = 2482 / 7467 = 0.332395875184
Terbukti bahwa hasil mendekati 0.33